internal/exporter/pulsarexporter/factor_test.go

```diff
@@ -25,16 +26,13 @@ func TestCreateDefaultConfig(t *testing.T) {
 	cfg := createDefaultConfig().(*Config)
 	assert.NotNil(t, cfg, "failed to create default config")
 	assert.NoError(t, configtest.CheckConfigStruct(cfg))
-	assert.Equal(t, defaultBroker, cfg.Brokers)
-	assert.Equal(t, "", cfg.Topic)
+	assert.Equal(t, defaultBroker, cfg.Broker)
+	assert.Equal(t, defaultMetricsTopic, cfg.Topic)
 }
 
 func TestCreateMetricsExport(t *testing.T) {
```

Comment:
I tried to run these tests locally but they tried to connect to a Pulsar instance and were unable. It's doable to create and run tests that connect to a Pulsar instance, but the tests should probably live in a separate file e.g. `integration_test.go` which is flagged as `integration` at the top, and they should start the containers they need. 

Here's a good example:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/apachereceiver/integration_test.go

I think that in order to exercise the code in factory.go without creating containers, the production code will have to be reworked a bit so that creating an exporter doesn't try to do any I/O. This will be a good change in my opinion, because we generally don't want to do I/O (e.g. connecting to Pulsar) until we Start() the exporter.